File: frontend/src/types/notifications.ts
// backend/app/schemas/notifications.py 1:1 대응 (docs/02-coding-conventions.md 9장)

export type NotificationType = "signal" | "exit" | "error";

export interface Notification {
  id: number;
  type: NotificationType;
  message: string;
  coin_symbol: string | null;
  is_read: boolean;
  created_at: string;
}

File: backend/app/schemas/notifications.py
"""04-settings 요청/응답 DTO — 알림 목록."""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class NotificationResponse(BaseModel):
    id: int
    type: Literal["signal", "exit", "error"]
    message: str
    coin_symbol: str | None
    is_read: bool
    created_at: datetime
